### PR TITLE
Add configurations to define ACLs for upload

### LINKED ISSUE #477: Disable creating/uploading episode ACL
I would like to have an option to disable episode ACL generation/ingest on my own hosted studio.

### LINKED ISSUE #478: Define an episode ACL (template) for ingest
I would like to configure the episode ACL in the `settings.json`. It can be a file/url path, where the template is located.

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Fixes #477 
Fixes #478 

A review from @wsmirnow would be appreciated to make sure this system is powerful enough for the customer systems you manage. Your adjustments [here](https://github.com/elan-ev/opencast-studio/commit/2e35c1592b557ae78f6bd2df0056a4dc662ab760) should be simple: just take the default ACL template and add the part you added in the linked commit. 

[This change](https://github.com/wsmirnow/opencast-studio/commit/c2b655ac35289baec8f542a7a6eb62852c9e1cb2) of yours was a bit more tricky as it required values taken from user roles. But I think with this PR, we can achieve the same effect with this ACL template:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Policy PolicyId="episodeACL"
  RuleCombiningAlgId="urn:oasis:names:tc:xacml:1.0:rule-combining-algorithm:permit-overrides"
  Version="2.0"
  xmlns="urn:oasis:names:tc:xacml:2.0:policy:schema:os">
  {{ #defaultReadRoles }}
    <Rule RuleId="{{ . }}_read_Permit" Effect="Permit">
      <Target>
        <Actions>
          <Action>
            <ActionMatch MatchId="urn:oasis:names:tc:xacml:1.0:function:string-equal">
              <AttributeValue DataType="http://www.w3.org/2001/XMLSchema#string">read</AttributeValue>
              <ActionAttributeDesignator AttributeId="urn:oasis:names:tc:xacml:1.0:action:action-id"
                DataType="http://www.w3.org/2001/XMLSchema#string"/>
            </ActionMatch>
          </Action>
        </Actions>
      </Target>
      <Condition>
        <Apply FunctionId="urn:oasis:names:tc:xacml:1.0:function:string-is-in">
          <AttributeValue DataType="http://www.w3.org/2001/XMLSchema#string">{{ . }}</AttributeValue>
          <SubjectAttributeDesignator AttributeId="urn:oasis:names:tc:xacml:2.0:subject:role"
            DataType="http://www.w3.org/2001/XMLSchema#string"/>
        </Apply>
      </Condition>
    </Rule>
  {{ /defaultReadRoles }}
  {{ #defaultWriteRoles }}
    <Rule RuleId="{{ . }}_write_Permit" Effect="Permit">
      <Target>
        <Actions>
          <Action>
            <ActionMatch MatchId="urn:oasis:names:tc:xacml:1.0:function:string-equal">
              <AttributeValue DataType="http://www.w3.org/2001/XMLSchema#string">write</AttributeValue>
              <ActionAttributeDesignator AttributeId="urn:oasis:names:tc:xacml:1.0:action:action-id"
                DataType="http://www.w3.org/2001/XMLSchema#string"/>
            </ActionMatch>
          </Action>
        </Actions>
      </Target>
      <Condition>
        <Apply FunctionId="urn:oasis:names:tc:xacml:1.0:function:string-is-in">
          <AttributeValue DataType="http://www.w3.org/2001/XMLSchema#string">{{ . }}</AttributeValue>
          <SubjectAttributeDesignator AttributeId="urn:oasis:names:tc:xacml:2.0:subject:role"
            DataType="http://www.w3.org/2001/XMLSchema#string"/>
        </Apply>
      </Condition>
    </Rule>
  {{ /defaultWriteRoles }}
  <Rule RuleId="DenyRule" Effect="Deny"/>
</Policy>
```

(Note that the `{{ #foo }} bar{{ . }}baz {{ /foo }}` syntax repeats `bar{{ . }}baz` for every element in `foo`, with `{{ . }}` replace by that element.)

### CODE REVIEW

**Comment by wsmirnow** [comment]
If you define  an ACL template block like this:
```
{{ #ltiCourseId }} <my acl here>{{ . }} <other block> {{ /ltiCourseId }}
```
and the user serve studio _not_ from an LTI session.  The `ltiCourseId` will be `undefined`. Will the block dissappear in the produced ACL or does the process fail?

**Comment by LukasKalbertodt** [comment]
> Will the block dissappear in the produced ACL or does the process fail?

@wsmirnow The block will disappear. See [this aprt of the Mustache documentation](https://github.com/janl/mustache.js/#false-values-or-empty-lists). So these `{{ #foo }} ... {{ /foo }}` blocks completely disappear if `foo` is `false`, `undefined`, an empty list or other "falsy" values.

File: README.md [hunk 1]
```diff
@@ -147,6 +148,74 @@ https://studio.opencast.org/?opencast.serverUrl=https://develop.opencast.org&upl
...
+
+The default ACL definition template simply gives read and write access to `userRole`:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<Policy PolicyId="mediapackage-1"
+  RuleCombiningAlgId="urn:oasis:names:tc:xacml:1.0:rule-combining-algorithm:permit-overrides"
+  Version="2.0"
+  xmlns="urn:oasis:names:tc:xacml:2.0:policy:schema:os">
+  <Rule RuleId="Administrator_read_Permit" Effect="Permit">
```
Comment (lkiesow) [inline README.md]:
```suggestion
  <Rule RuleId="user_read_Permit" Effect="Permit">
```

File: README.md [hunk 2]
```diff
@@ -147,6 +148,74 @@ https://studio.opencast.org/?opencast.serverUrl=https://develop.opencast.org&upl
...
+    </Target>
+    <Condition>
+      <Apply FunctionId="urn:oasis:names:tc:xacml:1.0:function:string-is-in">
+        <AttributeValue DataType="http://www.w3.org/2001/XMLSchema#string">{{ userRole }}</AttributeValue>
+        <SubjectAttributeDesignator AttributeId="urn:oasis:names:tc:xacml:2.0:subject:role"
+          DataType="http://www.w3.org/2001/XMLSchema#string"/>
+      </Apply>
+    </Condition>
+  </Rule>
+  <Rule RuleId="Administrator_write_Permit" Effect="Permit">
```
Comment (lkiesow) [inline README.md]:
```suggestion
  <Rule RuleId="user_write_Permit" Effect="Permit">
```

**Pull request merged**